internal/match/dict.go

```diff
@@ -0,0 +1,405 @@
+				w = w[:4]
+
+			case string(w) == "c" && lo > 0 && text[lo-1] == '(' && int(hi) < len(text) && text[hi] == ')':
+				w = copyright
+				lo--
+				hi++
+
+			case string(w) == "©":
+				w = copyright
+			}
+		}
+
+	Emit:
+		id, ok := d.dict[string(w)]
+		if ok {
+			if len(words) > 0 && words[len(words)-1].ID == id && string(w) == "copyright" {
+				// Treat "Copyright ©" as a single "copyright" instead of two.
+				continue
+			}
+			words = append(words, Word{id, lo, hi})
+			continue
+		}
+
+		if insert {
+			words = append(words, Word{d.Insert(string(w)), lo, hi})
+			continue
+		}
+
+		// Unknown word
+		words = append(words, Word{BadWord, lo, hi})
+	}
+
+	return words
+}
+
+var copyright = []byte("copyright")
+
+// foldRune returns the folded rune r.
+// It returns -1 if the rune r should be omitted entirely.
+//
+// Folding can be include canonicalizing transformation we want.
+// For now folding means:
+//	- fold to consistent case (unicode.SimpleFold, but moving to lower-case afterward)
+//	- return -1 for (drop) combining grave and acute U+0300, U+0301
+//	- strip pre-combined graves and acutes on vowels:
+//		é to e, etc (for Canadian or European licenses
+//		mentioning Québec or Commissariat à l'Energie Atomique)
+//
+// If necessary we could do a full Unicode-based conversion,
+// but that will require more thought about exactly what to do
+// and doing it efficiently. For now, the accents are enough.
+func foldRune(r rune) rune {
+	// Iterate SimpleFold until we hit the min equivalent rune,
+	// which - for the ones we care about - is the upper case ASCII rune.
+	for {
+		r1 := unicode.SimpleFold(r)
+		if r1 >= r {
+			break
+		}
+		r = r1
+	}
+
+	switch r {
+	case 'Á', 'À':
+		return 'a'
+	case 'É', 'È':
+		return 'e'
+	case 'Í', 'Ì':
+		return 'i'
+	case 'Ó', 'Ò':
+		return 'o'
+	case 'Ú', 'Ù':
+		return 'u'
+	}
+
+	if 'A' <= r && r <= 'Z' {
+		r += 'a' - 'A'
+	}
+
+	return r
+}
+
+// toFold converts s to folded form.
+func toFold(s string) string {
+	var buf []byte
+	for _, r := range s {
+		buf = appendFoldRune(buf, r)
+	}
+	return string(buf)
+}
+
+// appendFoldRune appends foldRune(r) to buf and returns the new buffer.
+func appendFoldRune(buf []byte, r rune) []byte {
+	r = foldRune(r)
+	if r < 0 {
+		return buf
+	}
+	if r < utf8.RuneSelf {
+		return append(buf, byte(r))
+	}
+
+	n := len(buf)
+	s := utf8.RuneLen(r)
+	for cap(buf) < n+s {
+		buf = append(buf[:cap(buf)], 0)
+	}
+	buf = buf[:n+s]
+	utf8.EncodeRune(buf[n:], r)
+	return buf
+}
+
+// isWordStart reports whether r can appear at the start of a word.
+func isWordStart(r rune) bool {
+	return unicode.IsLetter(r) || unicode.IsDigit(r) || r == '©'
+}
+
+// isWordContinue reports whether r can appear in a word, after the start.
+func isWordContinue(r rune) bool {
+	return unicode.IsLetter(r) || unicode.IsDigit(r) || unicode.Is(unicode.Mn, r)
+}
+
+// htmlTagSize returns the length of the HTML tag at the start of t, or else 0.
+func htmlTagSize(t string) int {
```

Comment:
this is tricky enough to deserve a distinct test. i know it's covered by the tests you have, but it's good practice to use unit tests when appropriate.